src/backends/beamcoder.ts

```diff
@@ -337,7 +332,10 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
                 if (!closestFrame) {
                     return outputFrame;
                 }
-                this.#filteredFrames = filteredFrames as beamcoder.DecodedFrames;
+                this.#filteredFrames = filteredFrames.concat(this.#filteredFrames);
+                while ((this.#filteredFrames as any).length > MAX_FILTERED_FRAME_COUNT) {
```

Comment:
We should work with packet frames instead of frames here. That way, we can keep track of which packet added which frames. That's important as `decoder.flush` spits out many frames. By keeping track of the packets, we only need to keep track of 2 or 3 packets instead of the current "random" amount.

Reply:
We should also add a test where we ask for the last frame multiple times